jupyter_server/base/handlers.py

```diff
@@ -134,9 +134,7 @@ def clear_login_cookie(self):
             self.force_clear_cookie(self.cookie_name)
 
     def get_current_user(self):
-        if self.login_handler is None:
-            return "anonymous"
-        return self.login_handler.get_user(self)
+        return self.identity_provider.get_user(self)
```

Comment:
Should we allow this method to fallback to using the login_handler is the identity_provider is a no-op to maintain backwards compatibility?

Reply:
If we decided to release this in a new major version, we can ignore this comment.